Set up for Documentation

LuisOlivaresJ opened the pull request:
1. New **docs** folder to found **Documentation** structure
* ``docs/source/index.rst`` is the root document (root of the “table of contents tree” or **toctree**)
* ``README.rst`` continues to be the welcome page, but using a new format ([reStructuredText](https://www.sphinx-doc.org/en/master/usage/restructuredtext/index.html))
* ``docs/source/installation.rst`` for installation instructions
* The main module overview is imported from the corresponding ``Docstring`` (E. g. from src/omg_dosimetry/calibration.py docstring) 

2. To do: import our documentation
2.1 [Create an account](https://readthedocs.org/accounts/signup/) on Read The Docs
2.2 Sign up or log in at Read the Docs with your GitHub account
2.3  Import our documentation repository ([here is a guide](https://docs.readthedocs.io/en/stable/intro/import-guide.html#importing-your-documentation)). Just for meantime, I suguest to use the "Documentation" branch instead the base "OMG_master"
2.4 On Administration, include me as a Maintainer

LuisOlivaresJ committed:
New folder for html documentation

- New ``docs`` folder for Sphinx documentation following this [QuickStartGuide](https://www.sphinx-doc.org/en/master/usage/quickstart.html)

- README format update to rst [reStructuredText](https://www.sphinx-doc.org/en/master/usage/restructuredtext/index.html) for an easy documentation incorporation

LuisOlivaresJ committed:
ReadTheDocs yml file

LuisOlivaresJ committed:
Merge pull request #1 from jfcabana/Documentation

ReadTheDocs yml file

LuisOlivaresJ committed:
Update .readthedocs.yml

LuisOlivaresJ committed:
Merge pull request #2 from jfcabana/Documentation

Update .readthedocs.yml

LuisOlivaresJ committed:
Delete requirements.txt

LuisOlivaresJ committed:
Merge pull request #3 from jfcabana/Documentation

Delete requirements.txt

LuisOlivaresJ committed:
Update requirements.txt

LuisOlivaresJ committed:
Update conf.py

LuisOlivaresJ committed:
Update conf.py

LuisOlivaresJ committed:
requirements file name and path

LuisOlivaresJ committed:
requirements_read_the_docs new file

LuisOlivaresJ committed:
Update requirements_read_the_docs.txt

sphinx_contributors

LuisOlivaresJ committed:
new requirements file

LuisOlivaresJ committed:
Update installation.rst

LuisOlivaresJ committed:
Update conf.py

LuisOlivaresJ committed:
Merge pull request #4 from LuisOlivaresJ/Documentation

Documentation

jfcabana commented on docs/source/installation.rst:
```diff
@@ -0,0 +1,80 @@
+I'm new to Python
+=================
+
+That's okay! If you're not a programmer at all you'll have a few things to do to get up. Using pylinac requires not just the base language Python, but a few dependencies as well.
```
Should change references to pylinac.

jfcabana commented on docs/source/installation.rst:
```diff
@@ -0,0 +1,80 @@
+
+That's okay! If you're not a programmer at all you'll have a few things to do to get up. Using pylinac requires not just the base language Python, but a few dependencies as well.
+Since most physicists don't program, or if they do it's in MATLAB, this section will help jumpstart your use of not
+just pylinac but Python in general and all its wonderful goodness! Getting started with Python takes some work to
```
same here

jfcabana commented:
I followed the steps to setup the project on Read the Docs. The compilation failed. Seems to be related to the `requirements.txt `file?

![image](https://github.com/jfcabana/omg_dosimetry/assets/136498453/cdbfee1d-85e3-4983-b65f-cd5f66f6cfe3)

LuisOlivaresJ committed:
Rename requirements.txt to requirements_test.txt

To test if the file name "requirements.txt" is automatically used by Read The Docs, instead "requirements_read_the_docs.txt"

LuisOlivaresJ commented:
Fixed. It seems that Read The Docs was trying to automatically use "requirements.txt" instead "requierements_read_the_docs.txt". I just renamed the first one and it worked. I am considering to delete it.

Have you see the [output page](https://omg-dosimetry.readthedocs.io/en/latest/index.html)? It is beautiful :blush:

LuisOlivaresJ committed:
Update installation.rst

jfcabana commented:
Yep, looking good! 

I thought the docstrings from the three modules would appear in the docs, but that does not seem to be the case?

LuisOlivaresJ committed:
Update calibration.rst

Docstring added from LUT's class

LuisOlivaresJ committed:
Merge pull request #7 from LuisOlivaresJ/Documentation

Update calibration.rst

LuisOlivaresJ commented:
Do you mean the docs folder? 
For example, the content in [calibration.rst](https://github.com/jfcabana/omg_dosimetry/blob/Documentation/docs/source/calibration.rst) file, is going to be the information to be show for the Calibration module on the web page. 
This is how it works, with the addition of two lines of code (see #7), 

![Captura](https://github.com/jfcabana/omg_dosimetry/assets/41806420/0ba48ebc-2d17-4c75-aede-03c71252b6de)

All the docstring from LUT's class (from PyPi omg_dosimetry package) is going to be show on the [web page](https://omg-dosimetry.readthedocs.io/en/latest/calibration.html#omg_dosimetry.calibration.LUT).

jfcabana commented:
Yes, that is it! There are several classes and functions that I think would be a good idea to include in the documentation.

jfcabana reviewed (approved):
I will need to update some of that documentation, but we can merge this as is for now to get started.

LuisOlivaresJ commented:
Perfect. I'll work on installation.rst to take acount about the spyder requeriment.

LuisOlivaresJ committed:
Merge pull request #8 from jfcabana/OMG_master

Update pyproject.toml

LuisOlivaresJ committed:
Merge pull request #5 from jfcabana/Documentation

Documentation

LuisOlivaresJ committed:
Changes needed for a correct packaging

Changed references to pylinac

pyproject file update for:

* a correct Packaging
* Addition of homepage, repository page and bug-tracker
* Automatic spyder installation when omg_dosimetry has been installed using pip

LuisOlivaresJ committed:
Merge pull request #9 from LuisOlivaresJ/Documentation

Changes needed for a correct packaging
